pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/index/loader/ForwardIndexHandlerTest.java

```diff
@@ -0,0 +1,432 @@
+      } else {
+        tempStringRows[i] = "n" + i;
+        tempIntRows[i] = i;
+        tempLongRows[i] = (long) i;
+      }
+    }
+
+    for (int i = 0; i < rowLength; i++) {
+      GenericRow row = new GenericRow();
+
+      // Raw String columns
+      row.putValue(SNAPPY_STRING, tempStringRows[i]);
+      row.putValue(ZSTANDARD_STRING, tempStringRows[i]);
+      row.putValue(PASS_THROUGH_STRING, tempStringRows[i]);
+      row.putValue(LZ4_STRING, tempStringRows[i]);
+
+      // Raw integer columns
+      row.putValue(SNAPPY_INTEGER, tempIntRows[i]);
+      row.putValue(ZSTANDARD_INTEGER, tempIntRows[i]);
+      row.putValue(PASS_THROUGH_INTEGER, tempIntRows[i]);
+      row.putValue(LZ4_INTEGER, tempIntRows[i]);
+
+      // Raw long columns
+      row.putValue(SNAPPY_LONG, tempLongRows[i]);
+      row.putValue(ZSTANDARD_LONG, tempLongRows[i]);
+      row.putValue(PASS_THROUGH_LONG, tempLongRows[i]);
+      row.putValue(LZ4_LONG, tempLongRows[i]);
+
+      // Dictionary columns
+      row.putValue(DICT_INTEGER, tempIntRows[i]);
+      row.putValue(DICT_LONG, tempLongRows[i]);
+      row.putValue(DICT_STRING, tempStringRows[i]);
+
+      rows.add(row);
+    }
+    return rows;
+  }
+
+  @Test
+  public void testComputeOperation() throws Exception {
+    // TEST1 : Validate with zero changes.
+    IndexLoadingConfig indexLoadingConfig = new IndexLoadingConfig(null, _tableConfig);
+    ForwardIndexHandler fwdIndexHandler =
+        new ForwardIndexHandler(_existingSegmentMetadata, indexLoadingConfig);
+    Map<String, ForwardIndexHandler.Operation> operationMap = new HashMap<>();
+    operationMap = fwdIndexHandler.computeOperation(_writer);
+    assertEquals(operationMap, Collections.EMPTY_MAP);
+
+    // TEST2: Enable dictionary for a RAW_ZSTANDARD_INDEX_COLUMN
+    indexLoadingConfig = new IndexLoadingConfig(null, _tableConfig);
+    indexLoadingConfig.getNoDictionaryColumns().remove(ZSTANDARD_STRING);
+    fwdIndexHandler = new ForwardIndexHandler(_existingSegmentMetadata, indexLoadingConfig);
+    operationMap = fwdIndexHandler.computeOperation(_writer);
+    assertEquals(operationMap, Collections.EMPTY_MAP);
+
+    // TEST3: Disable dictionary
+    indexLoadingConfig = new IndexLoadingConfig(null, _tableConfig);
+    indexLoadingConfig.getNoDictionaryColumns().add(DICT_INTEGER);
+    fwdIndexHandler = new ForwardIndexHandler(_existingSegmentMetadata, indexLoadingConfig);
+    operationMap = fwdIndexHandler.computeOperation(_writer);
+    assertEquals(operationMap, Collections.EMPTY_MAP);
+
+    // TEST4: Add random index
+    indexLoadingConfig = new IndexLoadingConfig(null, _tableConfig);
+    indexLoadingConfig.getTextIndexColumns().add(DICT_INTEGER);
+    indexLoadingConfig.getTextIndexColumns().add(LZ4_INTEGER);
+    fwdIndexHandler = new ForwardIndexHandler(_existingSegmentMetadata, indexLoadingConfig);
+    operationMap = fwdIndexHandler.computeOperation(_writer);
+    assertEquals(operationMap, Collections.EMPTY_MAP);
```

Comment:
What exactly are we trying to test in 1-4 ? Is it that ForwardIndexHandler is a NO-OP ? Maybe add a comment. For a sec I thought, this PR adds support for adding dictionary as well based on comment at line 247

Reply:
Done.